### ISSUE TITLE
Nothing displayed when searching, hit count seems correct

### ISSUE THREAD

**Issue by ocrasborn** [Signbank/Global-signbank#893]
If I search in the gloss field in the header for regular expression \+ (glosses containing a plus character), I get 161 results for NGT I see in the line above the results, but nothing is actually displayed in the list.

**Comment by susanodd** [comment]
I can see that it is indeed doing this.
It correctly queries the glosses.
But the sorting routine changes the type of the query result from QuerySet to a List before it should be displayed.

**Comment by susanodd** [comment]
What's weird is that they show up in the Query View.

![search-menubar-plus-queryview](https://user-images.githubusercontent.com/24272544/221589465-9cdf2c08-e606-43b9-9915-eb3be8d32bf0.png)

**Comment by susanodd** [comment]
This has been fixed! It's live now!

**Comment by ocrasborn** [comment]
Not quite. At least on Safari and Chrome/Mac, nothing is displayed when I search, no matter what the search is. The hit count is listed, but nothing appears.

**Comment by ocrasborn** [comment]
Same on Windows, it turns out, Lianne reports for the windows browser (Edge?) as well as Chrome.

**Comment by susanodd** [comment]
I had this recently also.
The javascript needed to be reloaded.

**Comment by ocrasborn** [comment]
On the new test server this is no problem, btw

**Comment by susanodd** [comment]
@ocrasborn can you try this again? This is working for me.

It needs to reload the javascript files, so make sure no other windows have old javascript.

**Comment by ocrasborn** [comment]
@susanodd Nope, it doesn't work for me on either browser. Is the reloading something I need to do locally? How do I do that? Do we expect this to happen more often?

**Comment by susanodd** [comment]
I had to keep doing it. I logged out and closed the browser window. Then opened it again.
The browser downloads the javascript locally in the browser memory. So old versions get stuck there sometimes.
I revised the Lemma Lookahead code and that is used in the Search and List views. That file hadn't been updated for a very long time.

**Comment by susanodd** [comment]
If you go to the Add New Gloss (under menu bar) and then start clicking on the lemma fields, and reload if necessary, you will see them in the correct languages. An extra function was added to the` lemma_typeahead.js` file for the lookahead, to set the initial language in the radio buttons. That's the file that needs to be reloaded for the Gloss List, since there's a Add Gloss Form there, too.

**Comment by susanodd** [comment]
The normal gloss search wasn't showing any results. The count N of Total was shown correctly but no list was displayed.

I had this error too early this morning, too. But after reloading the page, etc. it started working again.

I noticed on the new server, it takes it quite a long time for the scroll menu to be displayed. Then it appears. (When you're in Detail View, it should show there.)
BUT when I click things in the scroll bar, it puts be back on Global. (It leaves the new server.)

**Comment by ocrasborn** [comment]
Everything's fine now on Chrome/Mac, but on Safari/Mac all queries I try lead to a hit count but not to a display of the actual hits in the pane below.

**Comment by susanodd** [comment]
@ocrasborn is it possible you have a second window open? Or are you also logged in in another browser?

We will try this out on @vanlummelhuizen 's Apple tomorrow.

It could be related to the previews not showing. Perhaps something could be messed up with Ajax since both servers are running. I don't know whether the new server has the same IP address underneath. (In another issue, the scroll bar of the new server links back to Global.)

Remember that before we started using Ajax it would take ages for the search to display. Now it displays gradually.
Perhaps the number of rows per page is too large as a default.

I was restarting the server quite a few times today with new deployments, so that could have caused interruptions. Perhaps I caused it to timeout or something. We were working to debug the video upload.

**Comment by susanodd** [comment]
This still works for me on Safari, @ocrasborn.

It could have something to do with that the javascript files seem to be at the end of the page. (When you look at the elements panel of the page source after opening it via the developer pull-down of Safari)
It might be loading the javascript after the ajax calls (table rows) rather than when the page is first loaded.
(This is a guess.)

**Comment by vanlummelhuizen** [comment]
@ocrasborn Can you send us an exact url that gives an error? We are looking at the website from Safari on my iMac, but we have no problems. Or could you enter the Safari web inspector (Option+Command+I) and see what happens (e.g. in the Network or Console tab) when you (re)load the page.

**Comment by susanodd** [comment]
@ocrasborn try again?

Of all the weird things, there was a closing </div> tag missing in the Django template file.
It's weird that all the other browsers worked. Perhaps (newer) Safari is the only one to parse it before showing it.

**Comment by susanodd** [comment]
If that still doesn't solve it, perhaps something to do with these messages in Firefox:

```

> Sommige cookies maken misbruik van het aanbevolen attribuut ‘SameSite’ 4
> Cookie ‘csrftoken’ heeft geen juiste attribuutwaarde voor ‘SameSite’. Binnenkort worden cookies zonder het attribuut ‘SameSite’ of met een ongeldige waarde behandeld als ‘Lax’. Dit betekent dat de cookie niet meer in de context van derde partijen wordt verzonden. Als uw applicatie afhankelijk is van de beschikbaarheid van deze cookie in zulke contexten, voeg dan het attribuut ‘SameSite=None’ toe. Lees https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite voor meer informatie over het attribuut ‘SameSite’ [search](https://signbank.cls.ru.nl/signs/search/?csrfmiddlewaretoken=U9YiUl8VpeDiqSwDH33evn5SlhGqknjQauD6WKLjK0rjpGYrvvVyBb9jecrc0Q6d&sortOrder=&search_type=sign&search=&glosssearch_ar=&lemma_ar=&keyword_ar=&glosssearch_xx=&lemma_xx=&keyword_xx=&glosssearch_ca=&lemma_ca=&keyword_ca=&glosssearch_zh=&lemma_zh=&keyword_zh=&glosssearch_nl=&lemma_nl=&keyword_nl=&glosssearch_en=&lemma_en=&keyword_en=&glosssearch_fa=&lemma_fa=&keyword_fa=&glosssearch_fr=&lemma_fr=&keyword_fr=&glosssearch_he=&lemma_he=&keyword_he=&glosssearch_id=&lemma_id=&keyword_id=&glosssearch_ja=&lemma_ja=&keyword_ja=&glosssearch_no=&lemma_no=&keyword_no=&useInstr=&morpheme=&hasComponentOfType=0&hasMorphemeOfType=0&weakdrop=0&weakprop=0&domhndsh%5B%5D=21&domhndsh_letter=0&domhndsh_number=0&subhndsh_letter=0&subhndsh_number=0&locVirtObj=&phonOth=&mouthG=&mouthing=&phonetVar=&repeat=0&altern=0&iconImg=&concConcSet=&relation=&hasRelation=&relationToForeignSign=&hasRelationToForeignSign=0&inWeb=0&isNew=0&hasvideo=0&hasothermedia=0&definitionContains=&defspublished=unspecified&createdBy=&createdAfter=&createdBefore=&useInstr=&excludeFromEcv=0&show_all=False)
> Cookie ‘sessionid’ heeft geen juiste attribuutwaarde voor ‘SameSite’. Binnenkort worden cookies zonder het attribuut ‘SameSite’ of met een ongeldige waarde behandeld als ‘Lax’. Dit betekent dat de cookie niet meer in de context van derde partijen wordt verzonden. Als uw applicatie afhankelijk is van de beschikbaarheid van deze cookie in zulke contexten, voeg dan het attribuut ‘SameSite=None’ toe. Lees https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite voor meer informatie over het attribuut ‘SameSite’ [search](https://signbank.cls.ru.nl/signs/search/?csrfmiddlewaretoken=U9YiUl8VpeDiqSwDH33evn5SlhGqknjQauD6WKLjK0rjpGYrvvVyBb9jecrc0Q6d&sortOrder=&search_type=sign&search=&glosssearch_ar=&lemma_ar=&keyword_ar=&glosssearch_xx=&lemma_xx=&keyword_xx=&glosssearch_ca=&lemma_ca=&keyword_ca=&glosssearch_zh=&lemma_zh=&keyword_zh=&glosssearch_nl=&lemma_nl=&keyword_nl=&glosssearch_en=&lemma_en=&keyword_en=&glosssearch_fa=&lemma_fa=&keyword_fa=&glosssearch_fr=&lemma_fr=&keyword_fr=&glosssearch_he=&lemma_he=&keyword_he=&glosssearch_id=&lemma_id=&keyword_id=&glosssearch_ja=&lemma_ja=&keyword_ja=&glosssearch_no=&lemma_no=&keyword_no=&useInstr=&morpheme=&hasComponentOfType=0&hasMorphemeOfType=0&weakdrop=0&weakprop=0&domhndsh%5B%5D=21&domhndsh_letter=0&domhndsh_number=0&subhndsh_letter=0&subhndsh_number=0&locVirtObj=&phonOth=&mouthG=&mouthing=&phonetVar=&repeat=0&altern=0&iconImg=&concConcSet=&relation=&hasRelation=&relationToForeignSign=&hasRelationToForeignSign=0&inWeb=0&isNew=0&hasvideo=0&hasothermedia=0&definitionContains=&defspublished=unspecified&createdBy=&createdAfter=&createdBefore=&useInstr=&excludeFromEcv=0&show_all=False)
> Cookie ‘_ga’ heeft geen juiste attribuutwaarde voor ‘SameSite’. Binnenkort worden cookies zonder het attribuut ‘SameSite’ of met een ongeldige waarde behandeld als ‘Lax’. Dit betekent dat de cookie niet meer in de context van derde partijen wordt verzonden. Als uw applicatie afhankelijk is van de beschikbaarheid van deze cookie in zulke contexten, voeg dan het attribuut ‘SameSite=None’ toe. Lees https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite voor meer informatie over het attribuut ‘SameSite’ [analytics.js:25:486](https://www.google-analytics.com/analytics.js)
> Cookie ‘_gid’ heeft geen juiste attribuutwaarde voor ‘SameSite’. Binnenkort worden cookies zonder het attribuut ‘SameSite’ of met een ongeldige waarde behandeld als ‘Lax’. Dit betekent dat de cookie niet meer in de context van derde partijen wordt verzonden. Als uw applicatie afhankelijk is van de beschikbaarheid van deze cookie in zulke contexten, voeg dan het attribuut ‘SameSite=None’ toe. Lees https://developer.mozilla.org/docs/Web/HTTP/Headers/Set-Cookie/SameSite voor meer informatie over het attribuut ‘SameSite’

```

**Comment by susanodd** [comment]
Another possible cause is that the ajax calls didn't test whether the video exists for the gloss row.
With all the datasets selected, then searching, there are many results. Firefox was actually stopping with fetching because there were so many errors of videos not being there. (With a missing video it tries to fetch /protected_media/ without any file after it, which gives an error.)

**Comment by susanodd** [comment]
After that patch, now it gives errors for when the video file is empty or apparently not parsed (thus merely testing whether the file exists is not sufficient to not cause a network error.)

`

> https://signbank.cls.ru.nl/dictionary/protected_media/glossvideo/IS_Whynot/CO/CONTROL(ASL)-8622.mp4        NS_ERROR_PARSED_DATA_CACHED

`

`

> https://signbank.cls.ru.nl/dictionary/protected_media/glossvideo/Kata%20Kolok/WO/WORM-8107.mp4       0 B

`

**Comment by susanodd** [comment]
Also, this is weird:



`

> Accept-Language | nl,en-US;q=0.7,en;q=0.3

`
Why is it using `en-US` rather than `en` ?
(My Firefox browser is Dutch)

**Comment by vanlummelhuizen** [comment]
The string in the *Search Gloss* field is interpreted as a regular expression. If you fill in `+` you get an error because a plus is a special character in a regular expression and a single plus is not a correct regular expression. (See #938).

So, to search for glosses containing a plus, one should enter `\+`, 'escape plus'. The result is an empty table as reported by @ocrasborn. 

The latter seems to be solved on the new server, and soon on the live server too. Therefore, I am closing this issue.
